### DISCUSSION TITLE
Does EntitiesBT "skim" over nodes that instantly return NodeState.Success/Fail?

### DISCUSSION THREAD

**Discussion by ghost** [quabug/EntitiesBT#176] [Q&A] [answered]
For example, for the following behavior tree

```
Root
 Selector
  Instant
  Instant
  Instant
  Instant
  Multiframe
  Instant
  Instant
```

, in one "frame", does the BehaviorTreeRoot update past the first 4 Instant nodes (assuming they all return NodeState.Fail), and then start executing Multiframe?

**Comment by quabug** [comment, marked as answer]
Yes, a [`SelectorNode`](https://github.com/quabug/EntitiesBT/blob/cce56661b11acbc1bf4df7ca8641dce3d0423b5e/Packages/essential/Runtime/Nodes/Composites/SelectorNode.cs#L12) will only return/stop once the child state became success or running during `Tick`.
